conda_smithy/lint_recipe.py

```diff
@@ -372,10 +374,21 @@ def lintify_meta_yaml(
 
 def _maintainer_exists(maintainer: str) -> bool:
     """Check if a maintainer exists on GitHub."""
-    return (
-        requests.get(f"https://api.github.com/users/{maintainer}").status_code
-        == 200
-    )
+    if "GH_TOKEN" in os.environ:
+        # use a token if we have one
+        gh = github.Github(auth=github.Auth.Token(os.getenv("GH_TOKEN")))
```

Comment:
Doing this every time doubles the number of requests. Put this into a cached function?